pkgs/applications/networking/mkchromecast/default.nix

```diff
@@ -0,0 +1,84 @@
+{ lib, fetchFromGitHub, python3Packages, makeWrapper, sox, flac, faac, lame
+, ffmpeg, vorbis-tools, nodejs, pulseaudio, alsaLib, alsaUtils, youtube-dl
+, opusTools, gstreamer
+, alsaSupport ? false
+, systemTray ? true
+, ffmpegSupport ? true
+, youtubeSupport ? true
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "mkchromecast";
+  version = "0.3.8.1";
+
+  src = fetchFromGitHub rec {
+    owner = "muammar";
+    repo = pname;
+    rev = version;
+    url = "https://github.com/${owner}/${pname}/archive/${version}.tar.gz";
+    sha256 = "0baxilfalndip3fx8cj03pjkdywv1v65k41srarcdrzvm8dqca9s";
+  };
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  propagatedBuildInputs = let py = python3Packages; in [
+    py.PyChromecast
+    py.psutil
+    py.mutagen
+    py.flask
+    py.netifaces
+    py.requests
+    vorbis-tools
+    sox
+    flac
+    faac
+    lame
+    opusTools
+    gstreamer
+    nodejs
+  ] ++ lib.optional alsaSupport [ alsaLib ffmpeg ]
+    ++ lib.optional (!alsaSupport) pulseaudio
+    ++ lib.optional systemTray py.pyqt5
+    ++ lib.optional ffmpegSupport ffmpeg
+    ++ lib.optional youtubeSupport youtube-dl;
```

Comment:
All non-Python packages don't belong in here and should (conditionally) be added to `$PATH`.

Reply:
I should've solved the conditional part now. Are there packages here that don't belong entirely in the PATH (and which ones)?

Reply:
Also cleaned up by using `lib.strings.makeSearchPathOutput` now